Avoid 'using' the two-argument Utilities::MPI::ConsensusAlgorithms::Interface::run() function.

As #18768 and #18769 show, we should avoid using the deprecated `ConsensusAlgorithms::Process` class even within our own code base. As shown in #18768, we use it in the following context:
```
In file included from /home/martin/deal/lto_build/a/check.cc:3:
In file included from /home/martin/deal/deal.II/include/deal.II/lac/sparse_matrix_tools.h:20:
/home/martin/deal/deal.II/include/deal.II/base/mpi_consensus_algorithms.h:268:13: warning: 'Process<std::vector<std::pair<unsigned int, std::vector<std::pair<unsigned int, double>>>>, std::tuple<>>' is deprecated [-Wdeprecated-declarations]
  268 |         run(Process<RequestType, AnswerType> &process, const MPI_Comm comm);
      |             ^
/home/martin/deal/deal.II/include/deal.II/base/mpi_consensus_algorithms.h:912:31: note: in instantiation of template class 'dealii::Utilities::MPI::ConsensusAlgorithms::Interface<std::vector<std::pair<unsigned int, std::vector<std::pair<unsigned int, double>>>>, std::tuple<>>' requested here
  912 |       class Selector : public Interface<RequestType, AnswerType>
      |                               ^
/home/martin/deal/deal.II/include/deal.II/base/mpi_consensus_algorithms.h:1225:16: note: in instantiation of template class 'dealii::Utilities::MPI::ConsensusAlgorithms::Selector<std::vector<std::pair<unsigned int, std::vector<std::pair<unsigned int, double>>>>, std::tuple<>>' requested here
 1225 |         return Selector<RequestType, AnswerType>().run(
      |                ^
/home/martin/deal/deal.II/include/deal.II/base/mpi_consensus_algorithms.h:1247:16: note: in instantiation of function template specialization 'dealii::Utilities::MPI::ConsensusAlgorithms::selector<std::vector<std::pair<unsigned int, std::vector<std::pair<unsigned int, double>>>>, std::tuple<>>' requested here
 1247 |         return selector<RequestType, EmptyType>(
      |                ^
/home/martin/deal/deal.II/include/deal.II/lac/sparse_matrix_tools.h:271:52: note: in instantiation of function template specialization 'dealii::Utilities::MPI::ConsensusAlgorithms::selector<std::vector<std::pair<unsigned int, std::vector<std::pair<unsigned int, double>>>>>' requested here
  271 |       dealii::Utilities::MPI::ConsensusAlgorithms::selector<T1>(
      |                                                    ^
/home/martin/deal/deal.II/include/deal.II/lac/sparse_matrix_tools.h:473:17: note: in instantiation of function template specialization 'dealii::SparseMatrixTools::internal::extract_remote_rows<double, dealii::SparseMatrix<double>, dealii::SparsityPattern>' requested here
  473 |       internal::extract_remote_rows<Number>(system_matrix,
      |                 ^
/home/martin/deal/lto_build/a/check.cc:18:22: note: in instantiation of function template specialization 'dealii::SparseMatrixTools::restrict_to_full_matrices<dealii::SparseMatrix<double>, dealii::SparsityPattern, double>' requested here
   18 |   SparseMatrixTools::restrict_to_full_matrices(mat, sp, indices, blocks);
      |                      ^
/home/martin/deal/deal.II/include/deal.II/base/mpi_consensus_algorithms.h:164:13: note: 'Process<std::vector<std::pair<unsigned int, std::vector<std::pair<unsigned int, double>>>>, std::tuple<>>' has been explicitly marked deprecated here
  164 |       class DEAL_II_DEPRECATED Process
      |             ^
/home/martin/deal/mpi_build/include/deal.II/base/config.h:286:30: note: expanded from macro 'DEAL_II_DEPRECATED'
  286 | #define DEAL_II_DEPRECATED [[deprecated]]
      |                              ^
1 warning generated.
```

Interestingly, the `run()` function that is deprecated takes two arguments, but if one scrolls through the file, it is not actually ever called -- I think this may be a compiler quirk. Specifically, I think this frame...
```
/home/martin/deal/deal.II/include/deal.II/base/mpi_consensus_algorithms.h:912:31: note: in instantiation of template class 'dealii::Utilities::MPI::ConsensusAlgorithms::Interface<std::vector<std::pair<unsigned int, std::vector<std::pair<unsigned int, double>>>>, std::tuple<>>' requested here
  912 |       class Selector : public Interface<RequestType, AnswerType>
      |                               ^
```
is the key: This isn't actually anywhere in the call chain, but we are simply instantiating the `Selector` class, which has this code:
```
      template <typename RequestType, typename AnswerType>
      class Selector : public Interface<RequestType, AnswerType>
      {
      public:
...
        // Import the declarations from the base class.
        using Interface<RequestType, AnswerType>::run;
```
and the `Interface` class looks like this:
```
      template <typename RequestType, typename AnswerType>
      class Interface
      {
      public:
        /**
         * Default constructor.
         */
        Interface() = default;

        /**
         * Destructor. Made `virtual` to ensure that one can work with
         * derived classes.
         */
        virtual ~Interface() = default;

        /**
         * Run the consensus algorithm and return a vector of process ranks
         * that have requested answers from the current process.
         *
         * This version of the run() function simply unpacks the functions
         * packaged in `process` and calls the version of the run() function
         * that takes a number of `std::function` arguments.
         *
         * @deprecated Instead of deriving a class from the Process base class and
         *   providing a corresponding object to this function,
         *   use the other run() function in this class that takes function
         *   objects as arguments.
         */
        DEAL_II_DEPRECATED
        std::vector<unsigned int>
        run(Process<RequestType, AnswerType> &process, const MPI_Comm comm);
```
So I *think* that the error message *actually* results from the compiler referencing the deprecated `run()` function in the `using` statement. We do this in all of the classes in this file.

I think that we should be able to avoid that using the fact that the `using` declaration imports both `run()` functions from the base class, but that all of the classes overload the other `run()` function. We can then simply replace the `using` declaration by providing the two-argument overload like so:
```
        DEAL_II_DEPRECATED
        std::vector<unsigned int>
        run(Process<RequestType, AnswerType> &process, const MPI_Comm comm)
        {
          return Interface<RequestType, AnswerType>::run (process, comm);
        }
```

@kronbichler Would you be able to test this? I think you basically just have to copy these six lines in all 4 places where you have
```
        using Interface<RequestType, AnswerType>::run;
```
in this place.

--- a/include/deal.II/base/mpi_consensus_algorithms.h
+++ b/include/deal.II/base/mpi_consensus_algorithms.h
@@ -329,8 +329,12 @@ namespace Utilities
          */
         virtual ~NBX() = default;
 
-        // Import the declarations from the base class.
-        using Interface<RequestType, AnswerType>::run;
+        DEAL_II_DEPRECATED
+        std::vector<unsigned int>
+        run(Process<RequestType, AnswerType> &process, const MPI_Comm comm)
+        {
+          return Interface<RequestType, AnswerType>::run(process, comm);
+        }
 
         /**
          * @copydoc Interface::run()
@@ -579,8 +583,12 @@ namespace Utilities
          */
         virtual ~PEX() = default;
 
-        // Import the declarations from the base class.
-        using Interface<RequestType, AnswerType>::run;
+        DEAL_II_DEPRECATED
+        std::vector<unsigned int>
+        run(Process<RequestType, AnswerType> &process, const MPI_Comm comm)
+        {
+          return Interface<RequestType, AnswerType>::run(process, comm);
+        }
 
         /**
          * @copydoc Interface::run()
@@ -794,8 +802,12 @@ namespace Utilities
          */
         Serial() = default;
 
-        // Import the declarations from the base class.
-        using Interface<RequestType, AnswerType>::run;
+        DEAL_II_DEPRECATED
+        std::vector<unsigned int>
+        run(Process<RequestType, AnswerType> &process, const MPI_Comm comm)
+        {
+          return Interface<RequestType, AnswerType>::run(process, comm);
+        }
 
         /**
          * @copydoc Interface::run()
@@ -922,8 +934,12 @@ namespace Utilities
          */
         virtual ~Selector() = default;
 
-        // Import the declarations from the base class.
-        using Interface<RequestType, AnswerType>::run;
+        DEAL_II_DEPRECATED
+        std::vector<unsigned int>
+        run(Process<RequestType, AnswerType> &process, const MPI_Comm comm)
+        {
+          return Interface<RequestType, AnswerType>::run(process, comm);
+        }
 
         /**
          * @copydoc Interface::run()

